### PR TITLE
Enable Clang-Cuda 15 for Release Builds in the CI

### PR DESCRIPTION
**Pull request by SimeonEhrig**
~Since we deactivate Clang-Cuda (see #1906 ), we updated Catch2. Therefore let's test it again.~

Because of a compiler bug, only builds in debug does not work. Enable Clang-Cuda 15 release builds.

### CODE REVIEW

**Review by j-stephan — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -65,19 +65,21 @@ Supported Compilers
...
+| [HIP](https://alpaka.readthedocs.io/en/latest/install/HIP.html) (clang)        | -                                           | -                                           | -                                           | :x:                    | :x:                                                    | :x:                                                     | :x:                                              | :x:                                         | :white_check_mark: (HIP 5.0 - 5.2)                | :white_check_mark: (HIP 5.3 - 5.4)     | :white_check_mark: (HIP 5.5) | :x:                | -                                                     | -                                    |
 
 Other compilers or combinations marked with :x: in the table above may work but are not tested in CI and are therefore not explicitly supported.
 
+\* due to a [compiler bug](https://github.com/llvm/llvm-project/issues/58491) in debug mode only release builds are supported
```
Comment (j-stephan) [inline README.md:81]: The `*` is easily overlooked. Maybe we can add `:heavy_exclamation_mark:` (renders as :heavy_exclamation_mark:) instead? Both here and in the matrix.

**Review by j-stephan — CHANGES_REQUESTED** [review]

**Review by psychocoderHPC — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (j-stephan) [reply README.md:81]: I can also do this in the clang-16 PR so we don't have to rerun the tests.

**Review by j-stephan — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (SimeonEhrig) [reply README.md:81]: I would prefer a classic foot notes. Is `[1]` a option?

Comment (j-stephan) [reply README.md:81]: Yes, footnotes would also work.

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**